@here/harp-geometry/lib/WrapPolygon.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EarthConstants, GeoCoordinates, webMercatorProjection } from "@here/harp-geoutils";
+import { Vector2, Vector3 } from "three";
+
+import { ClippingEdge } from "./ClipPolygon";
+
+class ClipEdge extends ClippingEdge {
+    constructor(
+        private readonly p0: [number, number],
+        private readonly p1: [number, number],
+        private readonly isInside: (p: Vector2) => boolean
+    ) {
+        super();
+    }
+
+    inside(point: Vector2, extent: number): boolean {
+        return this.isInside(point);
+    }
+
+    /**
+     * Computes the intersection of a line and this clipping edge.
+     *
+     * @remarks
+     * {@link https://en.wikipedia.org/wiki/Line%E2%80%93line_intersection
+     *    | line-line intersection}.
+     */
+    computeIntersection(a: Vector2, b: Vector2, extent: number): Vector2 {
+        const { x: x1, y: y1 } = a;
+        const { x: x2, y: y2 } = b;
+        const [x3, y3] = this.p0;
+        const [x4, y4] = this.p1;
+        const d = (x1 - x2) * (y3 - y4) - (y1 - y2) * (x3 - x4);
+        const px = ((x1 * y2 - y1 * x2) * (x3 - x4) - (x1 - x2) * (x3 * y4 - y3 * x4)) / d;
+        const py = ((x1 * y2 - y1 * x2) * (y3 - y4) - (y1 - y2) * (x3 * y4 - y3 * x4)) / d;
+        return new Vector2(px, py);
```

Comment:
line-line intersection is quite common, might be useful to have it in Math2D.ts, we already have there intersectLineAndCircle.